pkg/controller/cainjector/cainjector_suite_test.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2019 The Jetstack cert-manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cainjector_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	kscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/klogr"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	certmgrscheme "github.com/jetstack/cert-manager/pkg/client/clientset/versioned/scheme"
+)
+
+// TODO: this really should be done somewhere in cert-manager proper
+var (
+	scheme  = runtime.NewScheme()
+	cfg     *rest.Config
+	testenv *envtest.Environment
+)
+
+func init() {
+	kscheme.AddToScheme(scheme)
+	certmgrscheme.AddToScheme(scheme)
+}
+
+func TestServiceController(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Service Controller Suite")
+}
+
+var _ = BeforeSuite(func(done Done) {
+	logf.SetLogger(klogr.New())
+
+	testenv = &envtest.Environment{
+		CRDDirectoryPaths: []string{"../../../tmp/crds"},
```

Comment:
Where does these come from/are they required?

Reply:
this... was supposed to be removed, must've gotten messed up in a rebase.